pkg/k8s/watchers/watcher.go

```diff
@@ -231,35 +230,10 @@ var ciliumResourceToGroupMapping = map[string]watcherInfo{
 	synced.CRDResourceName(v2alpha1.CPIPName):           {skip, ""}, // Handled by multi-pool IPAM allocator
 }
 
-// resourceGroups are all of the core Kubernetes and Cilium resource groups
-// which the Cilium agent watches to implement CNI functionality.
-func resourceGroups(cfg WatcherConfiguration) (resourceGroups, waitForCachesOnly []string) {
-	k8sGroups := []string{
-		// To perform the service translation and have the BPF LB datapath
-		// with the right service -> backend (k8s endpoints) translation.
-		K8sAPIGroupServiceV1Core,
-
-		// Namespaces can contain labels which are essential for
-		// endpoints being restored to have the right identity.
-		k8sAPIGroupNamespaceV1Core,
-		// Pods can contain labels which are essential for endpoints
-		// being restored to have the right identity.
-		resources.K8sAPIGroupPodV1Core,
-		// To perform the service translation and have the BPF LB datapath
-		// with the right service -> backend (k8s endpoints) translation.
-		resources.K8sAPIGroupEndpointSliceOrEndpoint,
-	}
-
-	if cfg.K8sNetworkPolicyEnabled() {
-		// When the flag is set,
-		// We need all network policies in place before restoring to
-		// make sure we are enforcing the correct policies for each
-		// endpoint before restarting.
-		waitForCachesOnly = append(waitForCachesOnly, k8sAPIGroupNetworkingV1Core)
-	}
-
-	ciliumResources := synced.AgentCRDResourceNames()
+func GetGroupsForCiliumResources(ciliumResources []string) ([]string, []string) {
```

Comment:
I think this may be outside watcher's purview. I would imagine whoever starts the watcher can determine the resources.

Reply:
Any other component starting the watcher _does not_ need to use this conversion helper, rather, they will just be passing a function analogous to `allResourceGroups`

Also, we'd need to duplicate `ciliumResourceToGroupMapping` and potentially the logic from this function at each call site, if this was moved out of the watcher, where it currently is.